ouroboros-praos/src/consensus/mod.rs

```diff
@@ -0,0 +1,349 @@
+            error!(
+                "VRF vkey hash in block ({}) does not match registered ledger vrf vkey hash ({})",
+                hex::encode(vrf_vkey_hash),
+                hex::encode(ledger_vrf_vkey_hash)
+            );
+            return false;
+        }
+        true
+    }
+
+    fn mk_vrf_input(&self, absolute_slot: u64, eta0: &[u8]) -> Hash<32> {
+        trace!("mk_vrf_input() absolute_slot {}", absolute_slot);
+        let mut hasher = Hasher::<256>::new();
+        hasher.input(&absolute_slot.to_be_bytes());
+        hasher.input(eta0);
+        hasher.finalize()
+    }
+}
+
+impl Validator for BlockValidator<'_> {
+    fn validate(&self) -> bool {
+        self.validate_babbage_compatible()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::consensus::BlockValidator;
+    use ctor::ctor;
+    use mockall::predicate::eq;
+    use ouroboros::ledger::{MockPoolInfo, PoolId, PoolSigma};
+    use ouroboros::validator::Validator;
+    use pallas_crypto::hash::Hash;
+    use pallas_math::math::{FixedDecimal, FixedPrecision};
+    use pallas_traverse::MultiEraHeader;
+    use std::fs::File;
+    use std::io::Read;
+
+    #[ctor]
+    fn init() {
+        // set rust log level to TRACE
+        // std::env::set_var("RUST_LOG", "ouroboros-praos=trace");
+
+        // initialize tracing crate
+        tracing_subscriber::fmt::init();
+    }
+
+    #[test]
+    fn test_validate_conway_block() {
+        let mut pool_info = MockPoolInfo::new();
+        // bcsh pool in epoch 508 made block 10817298
+        let bcsh_pool_id = PoolId::from(
+            hex::decode("00beef0a9be2f6d897ed24a613cf547bb20cd282a04edfc53d477114")
+                .unwrap()
+                .as_slice(),
+        );
+        pool_info
+            .expect_sigma()
+            .with(eq(bcsh_pool_id))
+            .returning(|_| {
+                Ok(PoolSigma {
+                    numerator: 25626202470912,
+                    denominator: 22586623335121436,
+                })
+            });
+        pool_info
+            .expect_vrf_vkey_hash()
+            .with(eq(bcsh_pool_id))
+            .returning(|_| {
+                Ok(<[u8; 32]>::try_from(
+                    hex::decode("c0d1f9b040d2f6fd7fc8775d24753d6db4b697429f11404a6178a0a4a005867b")
+                        .unwrap(),
+                )
+                .unwrap())
+            });
+
+        let epoch_nonce: Hash<32> =
+            "c7937fc47fecbe687891b3decd71e904d1e129598aa3852481d295eea3ea3ada"
+                .parse()
+                .unwrap();
+        let active_slots_coeff: FixedDecimal =
+            FixedDecimal::from(5u64) / FixedDecimal::from(100u64);
+
+        let c = (FixedDecimal::from(1u64) - active_slots_coeff).ln();
+
+        let conway_block_tag: u8 = 6;
+        let mut cbor = Vec::new();
+        File::open("tests/data/mainnet_blockheader_10817298.cbor")
+            .unwrap()
+            .read_to_end(&mut cbor)
+            .unwrap();
```

Comment:
```suggestion
        include_bytes!("tests/data/mainnet_blockheader_10817298.cbor")
```

More concise and we get a compilation error if we move or rename the file instead of a runtime error when executing tests.